src/tests/ftest/get_remote_files.sh

```diff
@@ -0,0 +1,319 @@
+    ARCHIVE_DEST = ${ARCHIVE_DEST}
+    FILES_TO_PROCESS = ${FILES_TO_PROCESS}
+    CART_LOGTEST = ${CART_LOGTEST}
+    VERBOSE = ${VERBOSE}"
+}
+
+#######################################
+# Convert human readable size values to respective byte value.
+# Arguments:
+#   size: size value to convert to bytes. i.e. 5mb, 40gb, 6k, 10g
+# Returns:
+#   converted byte value.
+#######################################
+human_to_bytes() {
+    h_size=${1,,}
+
+    declare -A units=(
+        [b]=1
+        [kb]=$(python -c "print (2**10)")
+        [k]=$(python -c "print (2**10)")
+        [mb]=$(python -c "print (2**20)")
+        [m]=$(python -c "print (2**20)")
+        [gb]=$(python -c "print (2**30)")
+        [g]=$(python -c "print (2**30)")
+    )
+    if [[ "${h_size,,}" =~ ([0-9.]+)([kbmgt]+) ]]; then
+        val="${BASH_REMATCH[1]}"
+        unit="${BASH_REMATCH[2]}"
+        (( val = val*${units[${unit}]} ))
+        echo "${val}"
+    else
+        echo "Invalid value provided: ${1}"
+        return 1
+    fi
+}
+
+#######################################
+# List and tag files with a 'Y' or 'N' to classify
+# big files based on the threshold provided.
+# Arguments:
+#   logs_dir: path to local logs
+#   threshold: value to use as threshold to classify big files
+# Returns:
+#   stdout output listing the files prepended with a 'Y' or 'N' tag
+#######################################
+list_tag_files() {
+    # shellcheck disable=SC2016,SC1004,SC2086
+    awk_cmd='{ \
+    if (th <= $1){ \
+        print "Y:",$1,$2} \
+    else { \
+        print "N:",$1,$2}\
+    }'
+
+    # Convert to bytes and run
+    bts=$(human_to_bytes "${2}")
+    # shellcheck disable=SC2086
+    du -ab -d 1 ${1} | awk -v th="${bts}" "${awk_cmd}"
+}
+
+check_hw() {
+    dmesg | grep "Hypervisor detected"
+    return $?
+}
+
+compress_files() {
+    echo "Compressing files ..."
+    echo "find ${1} -maxdepth 0 -type f -size +1M -print0 | xargs -r0 lbzip2 -v"
+    # shellcheck disable=SC2086
+    find ${1} -maxdepth 0 -type f -size +1M -print0 | xargs -r0 lbzip2 -v
+    return $?
+}
+
+scp_files() {
+    rc=0
+    copied=()
+    echo "Archiving logs in ${1} to ${2}"
+    # shellcheck disable=SC2045,SC2086
+    for file in $(ls -d ${1})
+    do
+        echo "scp -r ${file} ${2}/<hostname>-${file##*/}"
+        if scp -r "${file}" "${2}"/"$(hostname -s)"-"${file##*/}"; then
+            copied+=("${file}")
+            if ! sudo rm -fr "${file}"; then
+                rc=1
+                echo "Error removing ${file}"
+            fi
+        else
+            echo "Failure to copy ${file} to ${2}"
+            rc=1
+        fi
+    done
+    echo Copied "${copied[@]:-no files}"
+    return ${rc}
+}
+
+get_cartlogtest_files() {
+    rc=0
+    echo "Running ${CART_LOGTEST_PATH} on ${FILES_TO_PROCESS}"
+    # shellcheck disable=SC2045,SC2086
+    for file in $(ls -d ${1})
+    do
+        if "${CART_LOGTEST_PATH}" "${file}" &> "${file}"_cart_testlog; then
```

Comment:
This command will fail when `${file}` is a directory.  I'd suggest testing if `${file}` is a file first, something like:
```suggestion
        if [ -f ${file} ]; then
            if "${CART_LOGTEST_PATH}" "${file}" &> "${file}"_cart_testlog; then
```

Reply:
Done